### PR TITLE
docs: added instructions for installation via Puppet

### PR DESCRIPTION
**Pull request by danielsreichenbach**
This adds a small cookbook entry for deploying Minio using [Puppet](https://puppet.com).

### CODE REVIEW

**Comment by NitishT** [comment]
@deekoder any feedback on this?

File: docs/setup-using-puppet.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+[Puppet][puppet] is an Open Source configuration management system, allowing for
+automated system provisioning.
+
+In this recipe you will learn how to deploy Minio using Puppet as standalone
+installation.
+
+## 1. Prerequsites
+
+This cookbook assumes that you have a working Puppet setup (at least an agent for
+running Puppet recipes).
```
Comment (deekoder) [inline docs/setup-using-puppet.md]: A link to where reader can get instructions to setup puppet would be very essential here.

File: docs/setup-using-puppet.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        },
+        'region' => 'us-east-1',
+        'browser' => 'on',
+    },
+}
+```
+
+## 4. Recipe details
+
+*   The Puppet module downloads a platform specific Minio release from the
```
Comment (deekoder) [inline docs/setup-using-puppet.md]: Can we add commands to show how to do it. Eg.  wget or curl this url. Pass config like this command. Readers can simply cut copy paste those commands to try it out.

File: docs/setup-using-puppet.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    archives on <https://dl.minio.io/server/minio/release/> e.g. for 64bit
+    Linux we use <https://dl.minio.io/server/minio/release/linux-amd64/archive/>
+*   downloaded files will be checked against the SHA256 sums available for each
+    release.
+*   you can pass in _any_ configuration value to `minio::configuration` in hash
+    form that is documented on <https://docs.minio.io/docs/minio-server-configuration-guide>
+    to override defaults, e.g. setting different credentials or logging options
+    for `production` and `development` environments.
+
+## 5. Getting support
```
Comment (deekoder) [inline docs/setup-using-puppet.md]: We can remove this whole heading & section. Support is standardized via slack so this is breaking our recipe format.

**Comment by deekoder** [comment]
Closing this PR for it to be resubmitted with updated request for changes.

**Pull request closed without merging**